src/Razor/src/Microsoft.AspNetCore.Razor.Language/Syntax/MarkupTagHelperStartTagSyntax.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Razor.Language.Legacy;
+
+namespace Microsoft.AspNetCore.Razor.Language.Syntax
+{
+    internal partial class MarkupTagHelperStartTagSyntax
+    {
+        // Copied directly from MarkupStartTagSyntax Children & GetLegacyChildren.
+
+        public SyntaxList<RazorSyntaxNode> Children => GetLegacyChildren();
+
+        private SyntaxList<RazorSyntaxNode> GetLegacyChildren()
+        {
+            // This method returns the children of this start tag in legacy format.
+            // This is needed to generate the same classified spans as the legacy syntax tree.
+            var builder = new SyntaxListBuilder(5);
+            var tokens = SyntaxListBuilder<SyntaxToken>.Create();
+            var context = this.GetSpanContext();
+
+            // We want to know if this tag contains non-whitespace attribute content to set the appropriate AcceptedCharacters.
+            // The prefix of a start tag(E.g '|<foo| attr>') will have 'Any' accepted characters if non-whitespace attribute content exists.
+            var acceptsAnyContext = new SpanContext(context.ChunkGenerator, SpanEditHandler.CreateDefault());
+            acceptsAnyContext.EditHandler.AcceptedCharacters = AcceptedCharactersInternal.Any;
+            var containsAttributesContent = false;
+            foreach (var attribute in Attributes)
+            {
+                if (!string.IsNullOrWhiteSpace(attribute.GetContent()))
+                {
+                    containsAttributesContent = true;
+                    break;
+                }
+            }
+
+            if (!OpenAngle.IsMissing)
+            {
+                tokens.Add(OpenAngle);
+            }
+            if (Bang != null)
```

Comment:
This had a bang still which was interesting (since a TagHelper can't have one). But left it here so it maintains a direct copy